hikey_debian.mk

```diff
@@ -0,0 +1,408 @@
+
+# FIXME:
+# "make clean" in xtest: fails if optee_os has been cleaned previously
+.PHONY: xtest-clean
+xtest-clean: xtest-clean-common
+	rm -rf $(OPTEE_TEST_OUT_PATH)
+
+.PHONY: xtest-patch
+xtest-patch: xtest-patch-common
+
+################################################################################
+# aes-pef
+################################################################################
+PERF_FLAGS := CROSS_COMPILE_HOST=$(CROSS_COMPILE_NS_USER) \
+	CROSS_COMPILE_TA=$(CROSS_COMPILE_S_USER) \
+	TA_DEV_KIT_DIR=$(OPTEE_OS_TA_DEV_KIT_DIR)
+
+aes-perf: optee-os optee-client
+	$(MAKE) -C -j$(NPROC) $(AESPERF_PATH) $(PERF_FLAGS)
+
+.PHONY: aes-perf-clean
+aes-perf-clean:
+	rm -rf $(AESPERF_PATH)/out
+
+################################################################################
+# sha-perf
+################################################################################
+sha-perf: optee-os optee-client
+	$(MAKE) -C -j$(NPROC) $(SHAPERF_PATH) $(PERF_FLAGS)
+
+.PHONY: sha-perf-clean
+sha-perf-clean:
+	rm -rf $(SHAPERF_PATH)/out
+
+################################################################################
+# Boot Image
+################################################################################
+.PHONY: boot-img
+boot-img:
+ifeq ("$(wildcard $(BOOT_IMG))","")
+	echo "Downloading Debian HiKey boot image ..."
+	wget https://builds.96boards.org/snapshots/hikey/linaro/debian/410/boot-fat.uefi.img.gz -O $(BOOT_IMG).gz
+	gunzip $(BOOT_IMG).gz
+endif
+
+.PHONY: boot-img-clean
+boot-img-clean:
+	rm -f $(BOOT_IMG)
+
+################################################################################
+# system image
+#
+# TODO: This section is too hacky, in the future we should solve this, the
+# current issue is that, the developer edition uses an older 3.18 kernel and
+# lacks tools needed and the source.list for apt-get doesn't seem to be up to
+# date. On the other hand the "alip" seems to be working good, but that cannot
+# be downloaded using commandline tools, since there is a click through license
+# (and even if you could script that away, you're not allowed to do so, because
+# of the license).
+################################################################################
+.PHONY: system-img
+system-img: prepare
+ifeq ("$(wildcard $(SYSTEM_IMG))","")
+ifeq ($(DEVELOPER_EDITION),1)
+	@echo "Downloading Debian root fs (530MB) ..."
+	wget https://builds.96boards.org/snapshots/hikey/linaro/debian/410/hikey-jessie_developer_20160225-410.emmc.img.gz -O $(SYSTEM_IMG).gz
+	gunzip $(SYSTEM_IMG).gz
+else
+	@echo "Since there is a click through license, there is some manual work involved here."
+	@echo "The downloaded image, should be gunzip'd, renamed and placed here (don't change the name of the file)"
+	@echo "  $(SYSTEM_IMG)"
+	@read -r -p "Will spawn a new tab in your browser, press ENTER when ready to continue" dummy
+	@xdg-open https://builds.96boards.org/snapshots/hikey/linaro/debian/410/hikey-jessie_alip_20160225-410.emmc.img.gz
```

Comment:
My comment is badly worded and I will rewrite it. My intention was to say, that the gunzip'd file should have the name provided by the `@echo "  $(SYSTEM_IMG)"` line. I'll fix this.